state/remote/state.go

```diff
@@ -64,11 +65,20 @@ func (s *State) WriteStateForMigration(f *statefile.File, force bool) error {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	checkFile := statefile.New(s.state, s.lineage, s.serial)
+	// `force` is passed down from the CLI flag and terminates here. Actual
+	// force pushing with the remote backend happens when Put()'ing the contents
+	// in the backend. If force is specified we skip verifications and hand the
+	// context off to the client to use when persitence operations actually take place.
+	c, isForcePusher := s.Client.(ClientForcePusher)
 	if !force {
+		log.Print("[TRACE] state/remote: force push not specified")
+		checkFile := statefile.New(s.state, s.lineage, s.serial)
 		if err := statemgr.CheckValidImport(f, checkFile); err != nil {
 			return err
 		}
+	} else if isForcePusher {
```

Comment:
I'm a bit confused why the additional interface is needed, since there already is this `bool` related to force. Wouldn't a general `else` be what is needed?

Reply:
Open to all suggestions on how to make the comment in L68-L71 make more sense 🙏 

The CLI snags the flag and drops it in this function where the `s.Client` is just the `Client` [interface](https://github.com/hashicorp/terraform/blob/c18343eac28d6aa8dc744f06fbb662a4e60a2cc4/state/remote/remote.go#L12-L16) so L72 checks to see if the optional `ClientForcePusher` is implemented.

Would you rather change it to inline the check and remove the prior check and descriptive `isForcePusher` for the if-ok pattern?

```
else if c, ok := s.Client.(ClientForcePusher); ok {
	c.EnableForcePush()
}
```